packages/utilities/web-vitals/src/index.js

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from 'react';
+import useEvent from '@bbc/use-event';
+import { getCLS, getFID, getLCP, getFCP, getTTFB } from 'web-vitals';
+
+const webVitalsBase = {
+  age: 0,
+  type: 'web-vitals',
+  url: 'current-page-url',
+};
+
+const vitals = { cls: null, fid: null, lcp: null, fcp: null, ttfb: null };
+
+const updateWebVitals = ({ name, value }) => {
+  const vitalName = name.toLowerCase();
+  vitals[vitalName] = value;
+};
+
+const setCurrentUrl = () => {
+  webVitalsBase.url = window.location.href;
+};
+
+const sendBeacon = async (rawBeacon, reportingEndpoint) => {
+  const beacon = JSON.stringify(rawBeacon);
+  if (navigator.sendBeacon) {
+    await navigator.sendBeacon(reportingEndpoint, beacon);
+  } else {
+    await fetch(reportingEndpoint, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/reports+json' },
+      body: beacon,
+    });
+  }
+};
+
+const useWebVitals = ({ enabled, reportingEndpoint, loggerCallback }) => {
+  let pageLoadTime;
+  const sendVitals = async event => {
+    event.preventDefault();
+    const pageExitTime = new Date();
+    const pageAge = pageExitTime - pageLoadTime;
+
+    const beacon = [{ ...webVitalsBase, age: pageAge, body: { ...vitals } }];
+
+    try {
+      await sendBeacon(beacon, reportingEndpoint);
+    } catch (error) {
+      loggerCallback('web_vitals_send_error', {
+        ...error,
+      });
+    }
+  };
+
+  if (enabled) {
```

Comment:
I would be inclined to remove this and let the calling app decide whether web-vitals is enabled or not.

Reply:
Yeah, this bit of code is definitely going to change. Main reason I thought I'd pass in enabled as a prop rather than leaving it to the app to decide to call the hook at all was down to the fact that we shouldn't be triggering hooks conditionally (but hooks can have conditional logic inside of them). I figured I'd probably move the logic for it further down since it's currently making it call useEvent and useEffect conditionally within the hook